staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo.go

```diff
@@ -231,9 +231,16 @@ func (r *RequestInfoFactory) NewRequestInfo(req *http.Request) (*RequestInfo, er
 		}
 
 		if opts.FieldSelector != nil {
-			if name, ok := opts.FieldSelector.RequiresExactMatch("metadata.name"); ok {
-				if len(path.IsValidPathSegmentName(name)) == 0 {
-					requestInfo.Name = name
+			if len(requestInfo.Name) == 0 {
+				if name, ok := opts.FieldSelector.RequiresExactMatch("metadata.name"); ok {
+					if len(path.IsValidPathSegmentName(name)) == 0 {
+						requestInfo.Name = name
+					}
+				}
+			}
+			if len(requestInfo.Namespace) == 0 {
+				if namespace, ok := opts.FieldSelector.RequiresExactMatch("metadata.namespace"); ok {
+					requestInfo.Namespace = namespace
```

Comment:
This *must* validate that this is a valid namespace name

This also means that we're assuming the backing storage honors metadata.namespace field selectors, which is true for everything built into kube-apiserver but may not be true for everything served by an aggregated server. This would let `/apis/v1/foos?fieldSelector=metadata.namespace=bar` authorize as being within namespace `bar`, get proxied to the backing server which doesn't understand fieldSelectors, and get a cluster-scoped list of `foos`

Reply:
This is super subtle...

@p0lyn0mial - I'm not against trying to push further this PR, but I wouldn't rely on that for the PR you need and just try to use the exact same logic that we use here in your PR.

Reply:
ok, i will close this PR.